kong/timing/hooks/dns.lua

```diff
@@ -0,0 +1,42 @@
+local _M = {}
+
+local timing
+
+local client = package.loaded["kong.resty.dns.client"]
+if not client then
+  client = require("kong.tools.dns")()
+end
+
+
+local function before_toip(qname, _port, _dnsCacheOnly, _try_list)
+  timing.enter_context("dns")
+  timing.enter_context(qname)
+  timing.enter_context("resolve")
```

Comment:
` timing.enter_context("resolve")` seems unnecessary. I noticed that you didnt add one contenxt entry like `enter_context("request")` for resty.http library, right? Considering symmetry, there should also be only 2 occurrences here. Unless you want to differentiate HTTP/DNS action between connection and request phases.

For resty.http:

```
  timing.enter_context("external_http")
  timing.enter_context(destination)
```

Reply:
forget it is before_connect. http has 3 entry.